src/EditorFeatures/Core/Remote/SolutionChecksumUpdater.cs

```diff
@@ -28,18 +28,25 @@ internal sealed class SolutionChecksumUpdater
     /// </summary>
     private readonly IGlobalOperationNotificationService? _globalOperationService;
 
+    private readonly IDocumentTrackingService _documentTrackingService;
+
     /// <summary>
     /// Queue to push out text changes in a batched fashion when we hear about them.  Because these should be short
     /// operations (only syncing text changes) we don't cancel this when we enter the paused state.  We simply don't
     /// start queuing more requests into this until we become unpaused.
     /// </summary>
-    private readonly AsyncBatchingWorkQueue<(Document? oldDocument, Document? newDocument)> _textChangeQueue;
+    private readonly AsyncBatchingWorkQueue<(Document oldDocument, Document newDocument)> _textChangeQueue;
```

Comment:
change this queue so that we don't bother adding to it if the doc is null, instead of filtering those out on the other end when we process th queue items.